diff -ru meshlab.old/src/common/pluginmanager.cpp meshlab/src/common/pluginmanager.cpp
--- meshlab.old/src/common/pluginmanager.cpp	2017-05-04 15:02:14.000000000 +0300
+++ meshlab/src/common/pluginmanager.cpp	2017-05-06 02:24:19.632210608 +0300
@@ -192,7 +192,11 @@
 
 QString PluginManager::getBaseDirPath()
 {
+#if defined(Q_OS_LINUX)
+  QDir baseDir("/usr/lib/meshlab");
+#else
   QDir baseDir(qApp->applicationDirPath());
+#endif
 
 #if defined(Q_OS_WIN)
   // Windows:
