diff -ru meshlab.old/src/plugins_unsupported/render_rm/rmmeshrender.cpp meshlab/src/plugins_unsupported/render_rm/rmmeshrender.cpp
--- meshlab.old/src/plugins_unsupported/render_rm/rmmeshrender.cpp	2017-05-04 15:02:14.000000000 +0300
+++ meshlab/src/plugins_unsupported/render_rm/rmmeshrender.cpp	2017-05-06 02:28:00.765811456 +0300
@@ -62,8 +62,12 @@
 }
 
 void RmMeshShaderRenderPlugin::initActionList() {
-
+#if defined(Q_OS_LINUX)
+    QDir shadersDir = QDir("/usr/share/meshlab/shaders");
+#else
 	QDir shadersDir = QDir(qApp->applicationDirPath());
+#endif
+
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" ||
 	    shadersDir.dirName() == "release" ||
