diff -ru vcglib.old/wrap/callback.h vcglib/wrap/callback.h
--- vcglib.old/wrap/callback.h	2017-08-31 14:18:16.000000000 +0300
+++ vcglib/wrap/callback.h	2017-09-04 14:20:14.107856120 +0300
@@ -26,6 +26,7 @@
 
 #include <stdarg.h>  // For va_start, etc.
 #include <memory>    // For std::unique_ptr
+#include <iostream>
 
 namespace vcg {
 /*! \brief This function allow lenghty algorithms to report progress and status.
