--- cups-filters-1.0.71/filter/pdftoopvp/oprs/OPRS.cxx.old	2015-07-14 11:23:20.954425734 +0200
+++ cups-filters-1.0.71/filter/pdftoopvp/oprs/OPRS.cxx	2015-07-14 11:20:17.626843665 +0200
@@ -239,9 +239,9 @@ SplashError OPRS::drawImage(SplashImageS
 			      int w, int h, SplashCoord *mat) {
     if (rasterMode) {
 #if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
-	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
+	return splash->drawImage(src,NULL,srcData,srcMode,srcAlpha,w,h,mat);
 #else
-	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
+	return splash->drawImage(src,NULL,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
 #endif
     } else {
 	return opvpSplash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
--- cups-filters-1.0.71/filter/pdftoopvp/OPVPOutputDev.old	2015-07-14 11:22:37.188213406 +0200
+++ cups-filters-1.0.71/filter/pdftoopvp/OPVPOutputDev.cxx	2015-07-14 11:20:40.087958838 +0200
@@ -1809,10 +1809,10 @@ void OPVPOutputDev::drawSoftMaskedImage(
   maskColor[0] = 0;
   maskSplash->clear(maskColor);
 #if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
-  maskSplash->drawImage(&imageSrc, &imgMaskData,
+  maskSplash->drawImage(&imageSrc, NULL, &imgMaskData,
 			splashModeMono8, gFalse, maskWidth, maskHeight, mat);
 #else
-  maskSplash->drawImage(&imageSrc, &imgMaskData,
+  maskSplash->drawImage(&imageSrc, NULL, &imgMaskData,
 			splashModeMono8, gFalse, maskWidth, maskHeight,
                         mat,gFalse);
 #endif
