diff -ru work.old/kernel/nv-procfs.c work/kernel/nv-procfs.c
--- work.old/kernel/nv-procfs.c	2015-08-20 02:00:07.000000000 +0300
+++ work/kernel/nv-procfs.c	2015-11-06 17:35:13.647704931 +0300
@@ -356,7 +356,8 @@
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
 
-    return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    return 0;
 }
 
 static ssize_t
@@ -552,7 +553,8 @@
     void *v
 )
 {
-    return seq_puts(s, s->private);
+    seq_puts(s, s->private);
+    return 0;
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
