diff -ru ethminer-9999.old/cmake/EthCompilerSettings.cmake ethminer-9999/cmake/EthCompilerSettings.cmake
--- ethminer-9999.old/cmake/EthCompilerSettings.cmake	2017-06-23 21:21:57.045637919 +0300
+++ ethminer-9999/cmake/EthCompilerSettings.cmake	2017-06-23 21:25:00.310631977 +0300
@@ -3,7 +3,7 @@
 # C++11 check and activation
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
 
-	set(CMAKE_CXX_FLAGS "-Wall -Wno-unknown-pragmas -Wextra -Wno-error=parentheses -pedantic${CMAKE_CXX_FLAGS}")
+	set(CMAKE_CXX_FLAGS "-Wall -Wno-unknown-pragmas -Wextra -Wno-error=parentheses -pedantic ${CMAKE_CXX_FLAGS}")
 	set(CMAKE_CXX_FLAGS_DEBUG          "-O0 -g -DETH_DEBUG")
 	set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG -DETH_RELEASE")
 	set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG -DETH_RELEASE")
diff -ru ethminer-9999.old/libethash-cuda/CMakeLists.txt ethminer-9999/libethash-cuda/CMakeLists.txt
--- ethminer-9999.old/libethash-cuda/CMakeLists.txt	2017-06-23 21:21:57.676637899 +0300
+++ ethminer-9999/libethash-cuda/CMakeLists.txt	2017-06-23 21:23:00.936635848 +0300
@@ -1,6 +1,6 @@
 find_package(CUDA REQUIRED)
 
-set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};--disable-warnings;--ptxas-options=-v;-use_fast_math;-lineinfo)
+set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};-std=c++11;--disable-warnings;--ptxas-options=-v;-use_fast_math;-lineinfo)
 
 list(APPEND CUDA_NVCC_FLAGS_RELEASE -O3)
 list(APPEND CUDA_NVCC_FLAGS_DEBUG -G)
