diff -ru openMVG-9999.old/src/openMVG/matching/CMakeLists.txt openMVG-9999/src/openMVG/matching/CMakeLists.txt
--- openMVG-9999.old/src/openMVG/matching/CMakeLists.txt	2017-08-28 16:32:02.362953846 +0300
+++ openMVG-9999/src/openMVG/matching/CMakeLists.txt	2017-08-28 16:32:26.426953066 +0300
@@ -23,7 +23,7 @@
 add_library(openMVG_matching
   ${matching_files_header}
   ${matching_files_cpp})
-target_link_libraries(openMVG_matching PRIVATE openMVG_features stlplus)
+target_link_libraries(openMVG_matching PRIVATE openMVG_features stlplus ${FLANN_LIBRARY})
 target_link_libraries(openMVG_matching PUBLIC Threads::Threads)
 set_target_properties(openMVG_matching PROPERTIES SOVERSION ${OPENMVG_VERSION_MAJOR} VERSION "${OPENMVG_VERSION_MAJOR}.${OPENMVG_VERSION_MINOR}")
 set_property(TARGET openMVG_matching PROPERTY FOLDER OpenMVG/OpenMVG)
diff -ru openMVG-9999.old/src/software/SfM/clustering/CMakeLists.txt openMVG-9999/src/software/SfM/clustering/CMakeLists.txt
--- openMVG-9999.old/src/software/SfM/clustering/CMakeLists.txt	2017-08-28 16:32:02.396953845 +0300
+++ openMVG-9999/src/software/SfM/clustering/CMakeLists.txt	2017-08-28 16:32:23.035953176 +0300
@@ -7,7 +7,8 @@
   openMVG_image
   openMVG_features
   openMVG_sfm
-  domset)
+  domset
+  ${FLANN_LIBRARY})
 
 set_property(TARGET openMVG_main_ComputeClusters PROPERTY FOLDER OpenMVG/software/clustering)
 install(TARGETS openMVG_main_ComputeClusters DESTINATION bin/)
diff -ru openMVG-9999.old/src/software/SfM/CMakeLists.txt openMVG-9999/src/software/SfM/CMakeLists.txt
--- openMVG-9999.old/src/software/SfM/CMakeLists.txt	2017-08-28 16:32:02.395953845 +0300
+++ openMVG-9999/src/software/SfM/CMakeLists.txt	2017-08-28 16:32:23.037953176 +0300
@@ -40,6 +40,7 @@
   openMVG_sfm
   stlplus
   vlsift
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_ListMatchingPairs main_ListMatchingPairs.cpp)
@@ -49,6 +50,7 @@
   openMVG_multiview
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_ComputeMatches main_ComputeMatches.cpp)
@@ -86,6 +88,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_GlobalSfM main_GlobalSfM.cpp)
@@ -95,6 +98,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_ConvertSfM_DataFormat main_ConvertSfM_DataFormat.cpp)
@@ -103,6 +107,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_FrustumFiltering main_FrustumFiltering.cpp)
@@ -111,6 +116,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_ComputeStructureFromKnownPoses main_ComputeStructureFromKnownPoses.cpp)
@@ -119,6 +125,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_ComputeSfM_DataColor main_ComputeSfM_DataColor.cpp)
@@ -128,6 +135,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 ADD_EXECUTABLE(openMVG_main_PointsFiltering main_PointsFiltering.cpp)
@@ -135,6 +143,7 @@
   openMVG_system
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # Installation rules
@@ -165,6 +174,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 
@@ -176,6 +186,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - View tracks per image pair
@@ -186,6 +197,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export undistorted images related to a sfm_data file
@@ -197,6 +209,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # installation rules
@@ -227,6 +240,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export a SfM openMVG scene to openMVS format
@@ -238,6 +252,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export a SfM openMVG scene to CMPMVS format
@@ -249,6 +264,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 
@@ -261,6 +277,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export a SfM openMVG scene to meshlab scene with rasters
@@ -272,6 +289,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export a SfM openMVG scene to mvs-texturing scene folder
@@ -283,6 +301,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export SfM openMVG camera scene as triangle meshes
@@ -293,6 +312,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 # - Export SfM openMVG scene to NVM (N-View Match) file
@@ -302,7 +322,9 @@
   openMVG_image
   openMVG_features
   openMVG_sfm
-  stlplus )
+  stlplus
+  ${FLANN_LIBRARY}
+  )
 
 # installation rules
 SET_PROPERTY(TARGET openMVG_main_openMVG2PMVS PROPERTY FOLDER OpenMVG/software)
@@ -357,6 +379,7 @@
   openMVG_features
   openMVG_sfm
   stlplus
+  ${FLANN_LIBRARY}
   )
 
 #installation rules
