diff -ru screencloud-1.3.1.old/cmake/modules/FindPythonQt_QtAll.cmake screencloud-1.3.1/cmake/modules/FindPythonQt_QtAll.cmake
--- screencloud-1.3.1.old/cmake/modules/FindPythonQt_QtAll.cmake	2018-04-02 14:20:04.098542503 +0300
+++ screencloud-1.3.1/cmake/modules/FindPythonQt_QtAll.cmake	2018-04-02 14:23:33.963429749 +0300
@@ -3,10 +3,20 @@
 # Sets PYTHONQT_QTALL_FOUND, PYTHONQT_QTALL_INCLUDE_DIR, PYTHONQT_QTALL_LIBRARY, PYTHONQT_QTALL_LIBRARIES
 #
 
+get_property(LIB64 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS)
 
-find_path(PYTHONQT_QTALL_INSTALL_DIR NAMES include/PythonQt/PythonQt_QtAll.h include/PythonQt/extensions/PythonQt_QtAll/PythonQt_QtAll.h DOC "Directory where PythonQt_QtAll.h was installed.")
-find_path(PYTHONQT_QTALL_INCLUDE_DIR PythonQt_QtAll.h PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt" "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt/extensions/PythonQt_QtAll" ${CMAKE_INCLUDE_PATH} ${CMAKE_INSTALL_PREFIX}/include ${CMAKE_INSTALL_PREFIX}/include/PythonQt ${CMAKE_INSTALL_PREFIX}/include/PythonQt/extensions/PythonQt_QtAll DOC "Path to the PythonQt_QtAll include directory")
-find_library(PYTHONQT_QTALL_LIBRARY NAMES PythonQt_QtAll QtPython_QtAll PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" DOC "The PythonQt_QtAll library.")
+if ("${LIB64}" STREQUAL "TRUE")
+    set(LIB_SUFFIX 64)
+else()
+    set(LIB_SUFFIX "")
+endif()
+
+find_path(PYTHONQT_QTALL_INSTALL_DIR NAMES include/PythonQt/PythonQt_QtAll.h include/PythonQt/extensions/PythonQt_QtAll/PythonQt_QtAll.h extensions/PythonQt_QtAll/PythonQt_QtAll.h DOC "Directory where PythonQt_QtAll.h was installed.")
+find_path(PYTHONQT_QTALL_INCLUDE_DIR PythonQt_QtAll.h PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt" "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt/extensions/PythonQt_QtAll" "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" ${CMAKE_INCLUDE_PATH} ${CMAKE_INSTALL_PREFIX}/include ${CMAKE_INSTALL_PREFIX}/include/PythonQt ${CMAKE_INSTALL_PREFIX}/include/PythonQt/extensions/PythonQt_QtAll DOC "Path to the PythonQt_QtAll include directory")
+file(GLOB PYTHONQT_QTALL_LIB_FILE LIST_DIRECTORIES false RELATIVE "${PYTHONQT_QTALL_INSTALL_DIR}/lib${LIB_SUFFIX}" "${PYTHONQT_QTALL_INSTALL_DIR}/lib${LIB_SUFFIX}/*PythonQt_QtAll-Qt[4-9]*.so")
+string(REGEX REPLACE "^lib(.+)\\.so$" "\\1" PYTHONQT_QTALL_LIB "${PYTHONQT_QTALL_LIB_FILE}")
+
+find_library(PYTHONQT_QTALL_LIBRARY NAMES PythonQt_QtAll QtPython_QtAll "${PYTHONQT_QTALL_LIB}" PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" "${PYTHONQT_QTALL_INSTALL_DIR}/lib" DOC "The PythonQt_QtAll library.")
 
 mark_as_advanced(PYTHONQT_QTALL_INSTALL_DIR)
 mark_as_advanced(PYTHONQT_QTALL_INCLUDE_DIR)
@@ -20,4 +30,3 @@
   set(PYTHONQT_QTALL_FOUND 1)
   set(PYTHONQT_QTALL_LIBRARIES ${PYTHONQT_QTALL_LIBRARY} ${PYTHONQT_QTALL_LIBUTIL})
 endif()
-
