diff -ru AltYo-debian-0.4_rc23-linvinus1.old/altyo_settings.vala AltYo-debian-0.4_rc23-linvinus1/altyo_settings.vala
--- AltYo-debian-0.4_rc23-linvinus1.old/altyo_settings.vala	2017-01-02 11:35:31.000000000 +0300
+++ AltYo-debian-0.4_rc23-linvinus1/altyo_settings.vala	2017-05-06 16:54:26.334114169 +0300
@@ -710,7 +710,7 @@
         view.get_cursor(out path,out s_column);
         if(store.get_iter(out iter,path))
         if(!store.iter_has_child(iter)){
-          store.remove(iter);
+//          store.remove(iter);
           if(store.get_iter(out iter,path))
             view.set_cursor(path,null,false);
           else if(path.prev())
@@ -744,7 +744,7 @@
         view.get_cursor(out path,out s_column);
         if(store.get_iter(out iter,path))
         if(!store.iter_has_child(iter)){
-          store.remove(iter);
+//          store.remove(iter);
           if(store.get_iter(out iter,path))
             view.set_cursor(path,null,false);
           else if(path.prev())
@@ -778,7 +778,7 @@
         view.get_cursor(out path,out s_column);
         if(store.get_iter(out iter,path))
         if(!store.iter_has_child(iter)){
-          store.remove(iter);
+//          store.remove(iter);
           if(store.get_iter(out iter,path))
             view.set_cursor(path,null,false);
           else if(path.prev())
@@ -965,7 +965,7 @@
     string msg="";
     uint line,pos;
     if(!this.check_css(B.buffer.text,ref S,out line,out pos)){
-      msg=_("in line %d  at position %d error:%s").printf(line,pos,S);
+      msg=_("in line %d  at position %d error:%s").printf((int)line,(int)pos,S);
       debug("on_check_css_button_activate %s",msg);
       TextIter where;
       B.buffer.get_iter_at_line_offset(out where,(int)line,(int)pos);
@@ -1633,7 +1633,7 @@
               string S="";
               uint line,pos;
               if(!this.check_css(s,ref S,out line,out pos)){
-                string msg=_("New style will not be saved!\nin line %d  at position %d\nerror:%s").printf(line,pos,S);
+                string msg=_("New style will not be saved!\nin line %d  at position %d\nerror:%s").printf((int)line,(int)pos,S);
                 debug("on config apply css error %s",msg);
                 this.ayobject.main_window.show_message_box(_("AltYo CSS style error"),msg);
               }else{//looks good
diff -ru AltYo-debian-0.4_rc23-linvinus1.old/altyo_window.vala AltYo-debian-0.4_rc23-linvinus1/altyo_window.vala
--- AltYo-debian-0.4_rc23-linvinus1.old/altyo_window.vala	2017-01-02 11:35:31.000000000 +0300
+++ AltYo-debian-0.4_rc23-linvinus1/altyo_window.vala	2017-05-06 16:54:26.334114169 +0300
@@ -2891,7 +2891,7 @@
       for (item_it = this.children_removed.last(); item_it != null; item_it = item_it.prev) {
         AYTab tab = item_it.data;
         tab.stop_remove_timer();//disable timer while popup shown
-        string s = ( tab.tbutton.tab_title != null ? tab.tbutton.tab_title : _("index %d").printf(tab.tbutton.tab_index) );
+        string s = ( tab.tbutton.tab_title != null ? tab.tbutton.tab_title : _("index %d").printf((int)tab.tbutton.tab_index) );
         menuitem = new Gtk.MenuItem.with_label ("%d: %s".printf(index,s));
         menuitem.activate.connect(()=>{
           debug("trying to restore tab %s",menuitem.label);
