diff -ru ../src.old/meshlabplugins/filter_csg/filter_csg.pro ./meshlabplugins/filter_csg/filter_csg.pro
--- ../src.old/meshlabplugins/filter_csg/filter_csg.pro	2017-07-04 16:19:49.000000000 +0300
+++ ./meshlabplugins/filter_csg/filter_csg.pro	2017-09-04 13:00:25.491856692 +0300
@@ -22,7 +22,7 @@
 win32-msvc2012:LIBS		 += ../../external/lib/win32-msvc2008/mpirxx.lib ../../external/lib/win32-msvc2008/mpir.lib
 win32-msvc2013:LIBS		 += ../../external/lib/win32-msvc2008/mpirxx.lib ../../external/lib/win32-msvc2008/mpir.lib
 win32-msvc2015:LIBS		 += ../../external/lib/win32-msvc2015/mpirxx.lib ../../external/lib/win32-msvc2015/mpir.lib
-linux-g++:INCLUDEPATH += ../../external/inc/linux-g++/mpir-2.4.0
-#linux-g++:LIBS += $$PWD/../../external/lib/linux-g++/libmpirxx.a $$PWD/../../external/lib/linux-g++/libmpir.a
-#linux-g++-64:INCLUDEPATH += $$PWD/../../external/inc/linux-g++/mpir-2.4.0
-#linux-g++-64:LIBS += $$PWD/../../external/lib/linux-g++-64/libmpirxx.a $$PWD/../../external/lib/linux-g++-64/libmpir.a
+unix:LIBS += -lmpirxx -lmpir
+linux-g++-:LIBS += -lmpirxx -lmpir
+linux-g++-32:LIBS += -lmpirxx -lmpir
+linux-g++-64:LIBS += -lmpirxx -lmpir
