diff -ru src.old/meshlabplugins/filter_func/filter_func.pro src/meshlabplugins/filter_func/filter_func.pro
--- src.old/meshlabplugins/filter_func/filter_func.pro	2017-07-04 16:19:49.000000000 +0300
+++ src/meshlabplugins/filter_func/filter_func.pro	2017-09-04 13:04:12.618856664 +0300
@@ -19,7 +19,7 @@
 win32-msvc2015:LIBS += ../../external/lib/win32-msvc2015/muparser.lib
 
 macx:LIBS            += $$MACLIBDIR/libmuparser.a
-linux-g++:LIBS       += $$PWD/../../external/lib/linux-g++/libmuparser.a
-linux-g++-32:LIBS       += $$PWD/../../external/lib/linux-g++-32/libmuparser.a
-linux-g++-64:LIBS       += $$PWD/../../external/lib/linux-g++-64/libmuparser.a
+linux-g++:LIBS       += -lmuparser
+linux-g++-32:LIBS       += -lmuparser
+linux-g++-64:LIBS       += -lmuparser
 win32-g++:LIBS		 += ../../external/lib/win32-gcc/libmuparser.a
