diff -ru src.old/common/pluginmanager.cpp src/common/pluginmanager.cpp
--- src.old/common/pluginmanager.cpp	2017-07-04 16:19:49.000000000 +0300
+++ src/common/pluginmanager.cpp	2017-09-04 13:17:09.693856572 +0300
@@ -192,7 +192,7 @@
 
 QString PluginManager::getBaseDirPath()
 {
-  QDir baseDir(qApp->applicationDirPath());
+  QDir baseDir("/usr/share/meshlab");
 
 #if defined(Q_OS_WIN)
   // Windows:
@@ -214,7 +214,7 @@
 
 QString PluginManager::getDefaultPluginDirPath()
 {
-  QDir pluginsDir(getBaseDirPath());
+  QDir pluginsDir("/usr/lib/meshlab");
   if(!pluginsDir.exists("plugins"))
     //QMessageBox::warning(0,"Meshlab Initialization","Serious error. Unable to find the plugins directory.");
     qDebug("Meshlab Initialization: Serious error. Unable to find the plugins directory.");
