diff -ru src.old/meshlabplugins/render_gdp/meshrender.cpp src/meshlabplugins/render_gdp/meshrender.cpp
--- src.old/meshlabplugins/render_gdp/meshrender.cpp	2017-07-04 16:19:49.000000000 +0300
+++ src/meshlabplugins/render_gdp/meshrender.cpp	2017-09-04 13:39:29.739856412 +0300
@@ -28,6 +28,7 @@
 #include <QGLWidget>
 #include <QTextStream>
 #include "../../meshlab/glarea.h"
+#include <common/pluginmanager.h>
 
 using namespace std;
 using namespace vcg;
@@ -38,7 +39,8 @@
 	qaNone->setCheckable(false);
 	actionList << qaNone;*/
 
-	QDir shadersDir = QDir(qApp->applicationDirPath());
+    QDir shadersDir = PluginManager::getBaseDirPath();
+
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release" || shadersDir.dirName() == "plugins")
 		shadersDir.cdUp();
diff -ru src.old/meshlabplugins/render_gdp/shaderDialog.cpp src/meshlabplugins/render_gdp/shaderDialog.cpp
--- src.old/meshlabplugins/render_gdp/shaderDialog.cpp	2017-07-04 16:19:49.000000000 +0300
+++ src/meshlabplugins/render_gdp/shaderDialog.cpp	2017-09-04 13:40:16.131856406 +0300
@@ -32,6 +32,8 @@
 #include <QCheckBox>
 #include <QPushButton>
 
+#include <common/pluginmanager.h>
+
 #define DECFACTOR 100000.0f
 
 using namespace vcg;
@@ -317,7 +319,7 @@
 void ShaderDialog::browseTexturePath(int i) {
 	QFileDialog fd(0,"Choose new texture");
 
-	QDir shadersDir = QDir(qApp->applicationDirPath());
+    QDir shadersDir = PluginManager::getBaseDirPath();
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release")
 		shadersDir.cdUp();
diff -ru src.old/plugins_experimental/render_splatpyramid/dialog.cpp src/plugins_experimental/render_splatpyramid/dialog.cpp
--- src.old/plugins_experimental/render_splatpyramid/dialog.cpp	2017-07-04 16:19:49.000000000 +0300
+++ src/plugins_experimental/render_splatpyramid/dialog.cpp	2017-09-04 13:41:03.043856401 +0300
@@ -3,6 +3,7 @@
 #include <QFileDialog>
 #include <QMessageBox>
 
+#include <common/pluginmanager.h>
 
 #define DECFACTOR 100000.0f
 
@@ -123,7 +124,7 @@
 void Dialog::browseTexturePath(int i) {
   QFileDialog fd(0,"Choose new texture");
 
-  QDir shadersDir = QDir(qApp->applicationDirPath());
+  QDir shadersDir = PluginManager::getBaseDirPath();
 #if defined(Q_OS_WIN)
   if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release")
 	shadersDir.cdUp();
diff -ru src.old/plugins_experimental/render_splatpyramid/splat_pyramid.cc src/plugins_experimental/render_splatpyramid/splat_pyramid.cc
--- src.old/plugins_experimental/render_splatpyramid/splat_pyramid.cc	2017-07-04 16:19:49.000000000 +0300
+++ src/plugins_experimental/render_splatpyramid/splat_pyramid.cc	2017-09-04 13:41:40.291856396 +0300
@@ -18,6 +18,8 @@
 #include <QTextStream>
 #include <wrap/gl/trimesh.h>
 
+#include <common/pluginmanager.h>
+
 //#include <QGLFramebufferObject>
 
 //#include "GL/glut.h"
@@ -85,7 +87,7 @@
 		objects[i].setRendererType( render_mode );
 
   	/// Sets the path to the meshlab's shaders directory
-  	QDir shadersDir = QDir(qApp->applicationDirPath());
+    QDir shadersDir = PluginManager::getBaseDirPath();
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release" || shadersDir.dirName() == "plugins"  )
 		shadersDir.cdUp();
