diff -ru src.old/meshlabplugins/filter_ssynth/filter_ssynth.pro src/meshlabplugins/filter_ssynth/filter_ssynth.pro
--- src.old/meshlabplugins/filter_ssynth/filter_ssynth.pro	2017-07-04 16:19:49.000000000 +0300
+++ src/meshlabplugins/filter_ssynth/filter_ssynth.pro	2017-09-04 13:21:26.283856541 +0300
@@ -18,6 +18,8 @@
 win32-msvc2015:LIBS += ../../external/lib/win32-msvc2015/ssynth.lib
 win32-g++:LIBS += ../../external/lib/win32-gcc/libssynth.a
 linux-g++:LIBS += ../../external/lib/linux-g++/libssynth.a
+linux-g++:LIBS += -lgomp
+QMAKE_CXXFLAGS+=-fopenmp
 macx:LIBS += ../../external/lib/macx64/libssynth.a
 TARGET = filter_ssynth
 TEMPLATE = lib
