diff -ru src.old/meshlab_full.pro src/meshlab_full.pro
--- src.old/meshlab_full.pro	2017-07-04 16:19:49.000000000 +0300
+++ src/meshlab_full.pro	2017-09-04 13:23:34.170856526 +0300
@@ -52,6 +52,7 @@
                 meshlabplugins/filter_trioptimize \
                 meshlabplugins/filter_unsharp \
                 meshlabplugins/filter_voronoi \
+                plugins_unsupported/filter_poisson \
 # Rendering and Decoration Plugins
                 meshlabplugins/render_gdp \
                 meshlabplugins/render_radiance_scaling \
