diff -ru PrusaSlicer-version_2.4.0-beta2.old/src/slic3r/Utils/OctoPrint.cpp PrusaSlicer-version_2.4.0-beta2/src/slic3r/Utils/OctoPrint.cpp
--- PrusaSlicer-version_2.4.0-beta2.old/src/slic3r/Utils/OctoPrint.cpp	2021-11-26 17:15:51.000000000 +0300
+++ PrusaSlicer-version_2.4.0-beta2/src/slic3r/Utils/OctoPrint.cpp	2021-11-28 01:31:10.617427056 +0300
@@ -8,6 +8,7 @@
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
 #include <boost/algorithm/string/predicate.hpp>
+#include <boost/nowide/convert.hpp>
 
 #include <wx/progdlg.h>
 
