diff -ru PrusaSlicer-version_2.4.0-alpha1.old/src/libslic3r/Format/3mf.cpp PrusaSlicer-version_2.4.0-alpha1/src/libslic3r/Format/3mf.cpp
--- PrusaSlicer-version_2.4.0-alpha1.old/src/libslic3r/Format/3mf.cpp	2021-09-01 17:08:18.000000000 +0300
+++ PrusaSlicer-version_2.4.0-alpha1/src/libslic3r/Format/3mf.cpp	2021-09-09 03:08:15.641051008 +0300
@@ -14,6 +14,7 @@
 
 #include <limits>
 #include <stdexcept>
+#include <optional>
 
 #include <boost/algorithm/string/classification.hpp>
 #include <boost/algorithm/string/split.hpp>
