diff -ru dssi-vst-9999.old/Makefile dssi-vst-9999/Makefile
--- dssi-vst-9999.old/Makefile	2017-01-15 22:30:50.549511358 +0300
+++ dssi-vst-9999/Makefile	2017-01-15 22:37:39.155699247 +0300
@@ -10,8 +10,8 @@
 PREFIX  ?= /usr/local
 
 BIN_DIR    = $(DESTDIR)$(PREFIX)/bin
-DSSI_DIR   = $(DESTDIR)$(PREFIX)/lib/dssi
-LADSPA_DIR = $(DESTDIR)$(PREFIX)/lib/ladspa
+DSSI_DIR   ?= $(DESTDIR)$(PREFIX)/lib/dssi
+LADSPA_DIR ?= $(DESTDIR)$(PREFIX)/lib/ladspa
 
 BUILD_FLAGS  = -O2 -ffast-math -fvisibility=hidden -fPIC -mtune=generic -msse -Wall -Ivestige $(CXX_FLAGS)
 BUILD_FLAGS += $(shell pkg-config --cflags alsa liblo zlib)
@@ -101,6 +101,8 @@
 	install -m 755 vsthost $(BIN_DIR)
 	install -m 755 dssi-vst.so $(DSSI_DIR)
 	install -m 755 dssi-vst.so $(LADSPA_DIR)
-	install -m 755 dssi-vst_gui $(DSSI_DIR)/dssi-vst
-	install -m 755 dssi-vst-scanner.exe dssi-vst-scanner.exe.so $(DSSI_DIR)/dssi-vst
-	install -m 755 dssi-vst-server.exe dssi-vst-server.exe.so $(DSSI_DIR)/dssi-vst
+	install -D -m 755 dssi-vst_gui $(DSSIDIR)/dssi-vst/dssi-vst_gui
+	install -D -m 755 dssi-vst-server.exe.so $(DSSIDIR)/dssi-vst/dssi-vst-server.exe.so
+	install -D -m 755 dssi-vst-server.exe $(DSSIDIR)/dssi-vst/dssi-vst-server
+	install -D -m 755 dssi-vst-scanner.exe.so $(DSSIDIR)/dssi-vst/dssi-vst-scanner.exe.so
+	install -D -m 755 dssi-vst-scanner.exe $(DSSIDIR)/dssi-vst/dssi-vst-scanner
