diff -ru robomongo-0.8.5.old/src/robomongo/gui/AppStyle.h robomongo-0.8.5/src/robomongo/gui/AppStyle.h
--- robomongo-0.8.5.old/src/robomongo/gui/AppStyle.h	2015-03-11 16:22:11.000000000 +0300
+++ robomongo-0.8.5/src/robomongo/gui/AppStyle.h	2015-03-13 12:57:37.420106159 +0300
@@ -12,8 +12,8 @@
         #include <QProxyStyle>
         typedef QProxyStyle OsStyle;
     #else
-        #include <QCleanlooksStyle>
-        typedef QCleanlooksStyle OsStyle;
+        #include <QCommonStyle>
+        typedef QCommonStyle OsStyle;
     #endif
 #endif
 
