diff -ru robomongo-0.8.5.old/src/third-party/mongodb/src/mongo/pch.h robomongo-0.8.5/src/third-party/mongodb/src/mongo/pch.h
--- robomongo-0.8.5.old/src/third-party/mongodb/src/mongo/pch.h	2015-01-05 06:33:26.000000000 +0300
+++ robomongo-0.8.5/src/third-party/mongodb/src/mongo/pch.h	2015-02-04 15:58:21.165191956 +0300
@@ -39,6 +39,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <signal.h>
+#include <stdint.h>
 
 #include "time.h"
 #include "string.h"
