Только в stepmania-5.9999/autoconf/m4: ax_define_dir.m4
Только в stepmania-5.9999: b
diff -ru stepmania-5.9999.old/configure.ac stepmania-5.9999/configure.ac
--- stepmania-5.9999.old/configure.ac	2017-09-01 23:39:28.166695461 +0300
+++ stepmania-5.9999/configure.ac	2017-09-01 23:42:31.079379085 +0300
@@ -195,6 +195,11 @@
 	CXXFLAGS="$CXXFLAGS -msse2"
 fi
 
+AC_ARG_WITH(extdatadir, AC_HELP_STRING([--with-extdatadir], [Search Songs etc. from the configured DATADIR]), with_extdatadr=$withval, with_extdatadir=no)
+if test "$with_extdatadir" = "yes"; then
+	AX_DEFINE_DIR([EXT_DATADIR], [datadir], [External datadir])
+fi
+
 AC_ARG_WITH(prof, AS_HELP_STRING([--with-prof],[Enable profiling]), with_prof=$withval, with_prof=no)
 if test "$with_prof" = "yes"; then
 	test "$DEFAULT_CFLAGS" = "yes" && CFLAGS="$CFLAGS -pg"
Только в stepmania-5.9999: configure.ac.orig
Только в stepmania-5.9999/src/arch/ArchHooks: ArchHooks_Unix.cpp.orig
Только в stepmania-5.9999/src/arch/ArchHooks: ArchHooks_Unix.cpp.rej
diff -ru stepmania-5.9999.old/src/arch/LoadingWindow/LoadingWindow_Gtk.cpp stepmania-5.9999/src/arch/LoadingWindow/LoadingWindow_Gtk.cpp
--- stepmania-5.9999.old/src/arch/LoadingWindow/LoadingWindow_Gtk.cpp	2017-09-01 23:39:28.789684400 +0300
+++ stepmania-5.9999/src/arch/LoadingWindow/LoadingWindow_Gtk.cpp	2017-09-01 23:42:31.080379067 +0300
@@ -1,4 +1,5 @@
 #include "global.h"
+#include "ProductInfo.h"
 #include "RageLog.h"
 #include "RageFileManager.h"
 #include "RageUtil.h"
Только в stepmania-5.9999/src/arch/LoadingWindow: LoadingWindow_Gtk.cpp.orig
Только в stepmania-5.9999/src/arch/LoadingWindow: LoadingWindow_Gtk.cpp.rej
diff -ru stepmania-5.9999.old/src/arch/LoadingWindow/LoadingWindow_GtkModule.cpp stepmania-5.9999/src/arch/LoadingWindow/LoadingWindow_GtkModule.cpp
--- stepmania-5.9999.old/src/arch/LoadingWindow/LoadingWindow_GtkModule.cpp	2017-09-01 23:39:28.789684400 +0300
+++ stepmania-5.9999/src/arch/LoadingWindow/LoadingWindow_GtkModule.cpp	2017-09-01 23:42:31.080379067 +0300
@@ -1,4 +1,5 @@
 #include "global.h"
+#include "ProductInfo.h"
 #include "LoadingWindow_GtkModule.h"
 #include "RageUtil.h"
 #include "RageSurface.h"
@@ -15,7 +16,19 @@
 extern "C" const char *Init( int *argc, char ***argv )
 {
 	// Need to use external library to load this image. Native loader seems broken :/
+#if defined(EXT_DATADIR)
+	gchar splash_image_path[50] = EXT_DATADIR;
+	gchar product[] = PRODUCT_FAMILY;
+	int i,plen = strlen(product);
+	for (i = 0; i < plen; i++)
+		product[i] = tolower(product[i]);
+	strcat(splash_image_path,"/");
+	strcat(splash_image_path,product);
+	strcat(splash_image_path,"/Data/splash.png");
+#else
 	const gchar *splash_image_path = "Data/splash.png";
+#endif
+
 	GtkWidget *vbox;
 
 	gtk_disable_setlocale();
