diff -ru stepmania-5.1.0a.old/StepmaniaCore.cmake stepmania-5.1.0a/StepmaniaCore.cmake
--- stepmania-5.1.0a.old/StepmaniaCore.cmake	2015-10-31 01:43:03.000000000 +0300
+++ stepmania-5.1.0a/StepmaniaCore.cmake	2016-03-05 20:58:38.617508605 +0300
@@ -48,16 +48,20 @@
 # Set up the linker flags for MSVC builds.
 configure_msvc_runtime()
 
-# Determine which projects can be compiled in.
-if (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
-  message(STATUS "Using GNU compiler.")
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
-    set(CAN_COMPILE_TESTS OFF)
-  else()
-    set(CAN_COMPILE_TESTS ON)
-  endif()
+if(BUILD_UNIT_TESTS)
+	# Determine which projects can be compiled in.
+	if (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+		message(STATUS "Using GNU compiler.")
+		if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
+			set(CAN_COMPILE_TESTS OFF)
+		else()
+			set(CAN_COMPILE_TESTS ON)
+		endif()
+	else()
+		set(CAN_COMPILE_TESTS ON)
+	endif()
 else()
-  set(CAN_COMPILE_TESTS ON)
+  set(CAN_COMPILE_TESTS OFF)
 endif()
 
 # Checks the standard include directories for c-style headers.
