diff -ru stepmania-5.9999.orig/src/CMakeData-gtk.cmake stepmania-5.9999/src/CMakeData-gtk.cmake
--- stepmania-5.9999.orig/src/CMakeData-gtk.cmake	2019-09-10 02:07:19.927190559 +0300
+++ stepmania-5.9999/src/CMakeData-gtk.cmake	2019-09-10 02:18:17.323343315 +0300
@@ -8,6 +8,7 @@
             "arch/LoadingWindow/LoadingWindow_GtkModule.h")
 
 sm_add_compile_flag("GtkModule" "-std=${SM_CPP_STANDARD}")
+sm_add_compile_flag("GtkModule" "-isystem /usr/include/harfbuzz")
 if(CMAKE_CXX_COMPILER MATCHES "clang")
   sm_add_compile_flag("GtkModule" "-stdlib=libc++")
   set_target_properties("GtkModule" PROPERTIES LINK_FLAGS "-stdlib=libc++")
