diff -ur xbmc-13.0-Gotham_r2.old/configure.in xbmc-13.0-Gotham_r2/configure.in
--- xbmc-13.0-Gotham_r2.old/configure.in    2014-05-08 13:46:50.200954365 +0400
+++ xbmc-13.0-Gotham_r2/configure.in    2014-05-08 13:50:27.382567812 +0400
@@ -855,7 +855,7 @@
 elif test "$target_platform" = "target_raspberry_pi"; then
   ARCH="arm"
   use_arch="arm"
-elif test "$use_arch" = "arm"; then
+elif false; then
   CFLAGS="$CFLAGS -mno-apcs-stack-check"
   CXXFLAGS="$CXXFLAGS -mno-apcs-stack-check"
   FFMPEG_EXTRACFLAGS=""
