diff -ru openvdb.old/python/pyGrid.h openvdb/python/pyGrid.h
--- openvdb.old/python/pyGrid.h	2015-01-15 02:26:54.000000000 +0300
+++ openvdb/python/pyGrid.h	2015-10-13 12:46:55.133109413 +0300
@@ -41,7 +41,8 @@
 #ifdef PY_OPENVDB_USE_NUMPY
 #define PY_ARRAY_UNIQUE_SYMBOL PY_OPENVDB_ARRAY_API
 #define NO_IMPORT_ARRAY // NumPy gets initialized during module initialization
-#include <arrayobject.h> // for PyArrayObject
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#include <numpy/arrayobject.h> // for PyArrayObject
 #include "openvdb/tools/MeshToVolume.h"
 #include "openvdb/tools/VolumeToMesh.h" // for tools::volumeToMesh()
 #endif
