diff -ru openvdb.old/python/pyOpenVDBModule.cc openvdb/python/pyOpenVDBModule.cc
--- openvdb.old/python/pyOpenVDBModule.cc	2016-09-08 14:47:16.476373166 +0300
+++ openvdb/python/pyOpenVDBModule.cc	2016-09-08 14:49:39.061607585 +0300
@@ -598,7 +598,7 @@
 
 #ifdef PY_OPENVDB_USE_NUMPY
     // Initialize NumPy.
-    import_array();
+    if (_import_array()) { };
 #endif
 
     using namespace openvdb::OPENVDB_VERSION_NAME;
