diff -ru waflib.old/Context.py waflib/Context.py
--- waflib.old/Context.py	2011-03-29 18:52:54.000000000 +0400
+++ waflib/Context.py	2015-10-13 12:18:17.418964615 +0300
@@ -60,11 +60,8 @@
 		except KeyError:
 			global run_dir
 			rd=run_dir
-		class node_class(waflib.Node.Node):
-			pass
-		self.node_class=node_class
+		self.node_class = type("Nod3", (waflib.Node.Node,), {})
 		self.node_class.__module__="waflib.Node"
-		self.node_class.__name__="Nod3"
 		self.node_class.ctx=self
 		self.root=self.node_class('',None)
 		self.cur_script=None
