diff -ru makehuman-1.1.2.old/makehuman/core/module3d.py makehuman-1.1.2/makehuman/core/module3d.py
--- makehuman-1.1.2.old/makehuman/core/module3d.py	2017-12-07 02:09:50.014510155 +0300
+++ makehuman-1.1.2/makehuman/core/module3d.py	2017-12-07 02:10:58.489159361 +0300
@@ -308,7 +308,8 @@
         fuvs = self.fuvs[self.face_mask]
         uv_idx = np.unique(fuvs.reshape(-1))
         inverse_uv_idx = - np.ones(self.texco.shape[0], dtype=np.int32)
-        inverse_uv_idx[uv_idx] = np.arange(self.texco.shape[0], dtype=np.int32)
+        interval = np.arange(self.texco.shape[0], dtype=np.int32)
+        inverse_uv_idx[uv_idx] = interval[uv_idx]
         for i in range(self.vertsPerPrimitive):
             fuvs[:,i] = inverse_uv_idx[fuvs[:,i]]
 
