diff -ru gimp-2.9.4.old/configure.ac gimp-2.9.4/configure.ac
--- gimp-2.9.4.old/configure.ac	2016-07-12 22:17:49.000000000 +0300
+++ gimp-2.9.4/configure.ac	2016-07-14 15:57:47.257995017 +0300
@@ -674,10 +674,10 @@
 fi
 AC_MSG_RESULT($have_glib_2_43)
 
-AC_PATH_PROG(GEGL, gegl, no)
-if test "x$GEGL" = xno; then
-  add_deps_error([gegl], [Could not find gegl in your PATH.])
-fi
+#AC_PATH_PROG(GEGL, gegl, no)
+#if test "x$GEGL" = xno; then
+#  add_deps_error([gegl], [Could not find gegl in your PATH.])
+#fi
 
 
 # Check for bind_textdomain_codeset, including -lintl if GLib brings it in.
