From a858864986b058230fc52907d8f306e6dff89e28 Mon Sep 17 00:00:00 2001
From: Hielke Christian Braun <hcb@unco.de>
Date: Mon, 19 Jun 2017 19:49:37 +0200
Subject: [PATCH] remove -flto flag

---
 cmake/EthCompilerSettings.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/EthCompilerSettings.cmake b/cmake/EthCompilerSettings.cmake
index f82a946ac8..c3631747eb 100644
--- a/cmake/EthCompilerSettings.cmake
+++ b/cmake/EthCompilerSettings.cmake
@@ -6,11 +6,11 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
 	if (ETH_SHARED)
 		set(GCC_SHARED_FLAGS "-DSHAREDLIB -fPIC")
 	endif(ETH_SHARED)
-	set(CMAKE_EXE_LINKER_FLAGS "-flto -Wl,--gc-sections ${CMAKE_EXE_LINKER_FLAGS}")
+	set(CMAKE_EXE_LINKER_FLAGS "-Wl,--gc-sections ${CMAKE_EXE_LINKER_FLAGS}")
 	set(CMAKE_AR "gcc-ar")
 	set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> qcs <TARGET> <LINK_FLAGS> <OBJECTS>")
 	set(CMAKE_CXX_ARCHIVE_FINISH true)
-	set(CMAKE_CXX_FLAGS "-flto -ffunction-sections -std=c++11 -Wall -Wno-unknown-pragmas -Wextra -Wno-error=parentheses -pedantic ${GCC_SHARED_FLAGS} ${CMAKE_CXX_FLAGS}")
+	set(CMAKE_CXX_FLAGS "-ffunction-sections -std=c++11 -Wall -Wno-unknown-pragmas -Wextra -Wno-error=parentheses -pedantic ${GCC_SHARED_FLAGS} ${CMAKE_CXX_FLAGS}")
 	set(CMAKE_CXX_FLAGS_DEBUG          "-O0 -g -DETH_DEBUG")
 	set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG -DETH_RELEASE")
 	set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG -DETH_RELEASE")
