diff -ru cpp-ethereum-9999.old/libethash-cuda/CMakeLists.txt cpp-ethereum-9999/libethash-cuda/CMakeLists.txt
--- cpp-ethereum-9999.old/libethash-cuda/CMakeLists.txt	2017-06-23 21:42:55.000597130 +0300
+++ cpp-ethereum-9999/libethash-cuda/CMakeLists.txt	2017-06-23 21:43:46.791595451 +0300
@@ -7,7 +7,7 @@
 file(GLOB SRC_LIST "*.cpp" "*.cu")
 file(GLOB HEADERS "*.h" "*.cuh")
 
-set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};--disable-warnings;--ptxas-options=-v;-use_fast_math;-lineinfo)
+set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};-Xcompiler;-D__CORRECT_ISO_CPP11_MATH_H_PROTO;--disable-warnings;--ptxas-options=-v;-use_fast_math;-lineinfo)
 
 LIST(APPEND CUDA_NVCC_FLAGS_RELEASE -O3)
 LIST(APPEND CUDA_NVCC_FLAGS_DEBUG -G)
