--- a/modules/printbackends/cups/gtkprintbackendcups.c	2012-07-16 06:33:25.000000000 +0400
+++ b/modules/printbackends/cups/gtkprintbackendcups.c	2012-08-05 13:32:30.486421759 +0400
@@ -1835,9 +1835,9 @@
 	    info->auth_info_required[i] = g_strdup (ippGetString (attr, i, NULL));
 	}
     }
-  else if (strcmp (attr->name, "number-up-default") == 0)
+  else if (strcmp (ippGetName (attr), "number-up-default") == 0)
     {
-      info->default_number_up = attr->values[0].integer;
+      info->default_number_up = ippGetInteger (attr, 0);
     }
   else
     {
