diff -Naur firefox-54.0b1.orig/browser/installer/package-manifest.in firefox-54.0b1/browser/installer/package-manifest.in
--- firefox-54.0b1.orig/browser/installer/package-manifest.in	2017-04-18 14:22:35.000000000 +0200
+++ firefox-54.0b1/browser/installer/package-manifest.in	2017-04-20 19:36:24.592036520 +0200
@@ -639,6 +639,7 @@
 
 ; [Default Preferences]
 ; All the pref files must be part of base to prevent migration bugs
+@RESPATH@/browser/@PREF_DIR@/all-gentoo.js
 @RESPATH@/browser/@PREF_DIR@/firefox.js
 @RESPATH@/browser/@PREF_DIR@/firefox-branding.js
 @RESPATH@/greprefs.js
