--- dlls/wined3d/surface.c.old	2012-12-10 15:32:42.017715645 +0400
+++ dlls/wined3d/surface.c	2012-12-10 15:34:14.666888343 +0400
@@ -5442,8 +5442,8 @@
 
     /* Early sort out of cases where no render target is used */
     if (!dstSwapchain && !srcSwapchain
-            && src_surface != device->fb.render_targets[0]
-            && dst_surface != device->fb.render_targets[0])
+            && !(src_surface->resource.usage & WINED3DUSAGE_RENDERTARGET)
+            && !(dst_surface->resource.usage & WINED3DUSAGE_RENDERTARGET))
     {
         TRACE("No surface is render target, not using hardware blit.\n");
         return WINED3DERR_INVALIDCALL;
